http-client/src/main/java/io/airlift/http/client/jetty/JettyIoPoolManager.java

```diff
@@ -0,0 +1,73 @@
+package io.airlift.http.client.jetty;
+
+import com.google.inject.Injector;
+import com.google.inject.Key;
+
+import javax.annotation.PreDestroy;
+import javax.inject.Inject;
+
+import java.lang.annotation.Annotation;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class JettyIoPoolManager
```

Comment:
Let's refactor this to be a top-level class first, as a separate commit. That will make it easier to see the changes needed for shared pool removal.

Reply:
Actually, I think this class can be replaced with a `JettyIoPoolProvider`. It will probably look similar to `DiscoveryExecutorProvider`.